### PR TITLE
CLN: timeseries in plotting clean up

### PR DESCRIPTION
**Pull request by charlesdong1991**
`format_timedelta_ticks` appears both in `timeseries.py` and `matplotlib.converter.py`, and there is no test. Then i look through the code base, it is only used in `matplotlib.converter.py` for `TimeSeries_TimedeltaFormatter`. So I remove the one in `timeseries.py` and add test for it.

- [ ] closes #xxxx
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_converter.py [hunk 1]
```diff
@@ -373,3 +373,22 @@ def test_convert_nested(self):
         r1 = self.pc.convert([data, data], None, self.axis)
         r2 = [self.pc.convert(data, None, self.axis) for _ in range(2)]
         assert r1 == r2
+
+
+class TestTimeDeltaConverter:
+    def setup_method(self, method):
```
Comment (TomAugspurger) [inline pandas/tests/plotting/test_converter.py]: We're trying to move away from the `setup_`-style of tests. Can you use the full name in the test?

Comment (charlesdong1991) [reply pandas/tests/plotting/test_converter.py]: you mean `timedelta_converter_method`? @TomAugspurger

File: pandas/tests/plotting/test_converter.py [hunk 2]
```diff
@@ -373,3 +373,21 @@ def test_convert_nested(self):
...
+        [
+            (0.0, 0, "00:00:00"),
+            (3972320000000, 1, "01:06:12.3"),
+            (713233432000000, 2, "8 days 06:07:13.43"),
+            (32423432000000, 4, "09:00:23.4320"),
+        ],
+    )
+    def test_format_timedelta_ticks(self, x, decimal, format_expected):
+        tdc = converter.TimeSeries_TimedeltaFormatter
+        result = tdc.format_timedelta_ticks(x, pos=None, n_decimals=decimal)
```
Comment (jbrockmendel) [inline pandas/tests/plotting/test_converter.py]: would it be worthwhile to parametrize over `pos` values here?  or would that obscure the logic we actually care about testing?

Comment (charlesdong1991) [reply pandas/tests/plotting/test_converter.py]: pos is not being used in `format_timdelta_ticks` function, honestly i am not sure why/how this is used. So i just keep as is for now. @jbrockmendel

**Comment by charlesdong1991** [comment]
do you prefer to remove `pos`? @jbrockmendel or parametrize it?

File: pandas/tests/plotting/test_converter.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/plotting/test_converter.py]: Leaving it as is sounds fine, thanks for looking into it.

**Comment by charlesdong1991** [comment]
fancy a merge? :D  @TomAugspurger

**Comment by jreback** [comment]
lgtm. can you merge master and ping on green.

**Comment by charlesdong1991** [comment]
ping

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @charlesdong1991

**Pull request merged**